airflow/providers/google/cloud/hooks/cloud_sql.py

```diff
@@ -964,7 +964,7 @@ def create_connection(self) -> Connection:
         proxy, TCP, UNIX sockets, SSL.
         """
         uri = self._generate_connection_uri()
-        connection = Connection(conn_id=self.db_conn_id, uri=uri)
+        connection = Connection.from_uri(conn_id=self.db_conn_id, uri=uri)
```

Comment:
You can check the Airflow version or check if `Connection .__dict__ .get("from_uri")` is not None;  if the check fails, we need to fallback to old way (via constructor). Also we need to add a small todo comment to remove it when min_ariflow_version in the providers is >= your checked version.

Reply:
yeah we do that in several places for example:
https://github.com/apache/airflow/blob/49108e15eb2eb30e2ccb95c9332db7b38d35f2de/airflow/providers/cncf/kubernetes/executors/kubernetes_executor.py#L66-L74